tests/PHPStan/Analyser/data/bug-4714.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Bug4714;
+
+class Foo
+{
+	public function bar(\PDO $database): void
+	{
+		$statement = $database->prepare('SELECT `col` FROM `foo` WHERE `param` = :param');
+		$statement->bindParam(':param', $param);
+
+		$param = 1;
+
+		$statement->execute();
+		$statement->bindColumn('col', $col);
```

Comment:
Have you first made sure that the test fails? It won't because it's not referenced from anywhere. See `NodeScopeResolver::testFileAsserts`. The test needs to contain `assertVariableCertainty` call.

Reply:
Nope... took me 2 hours to figure out how to run a single test-method in isolation. I now made sure the test fails.

Reply:
What was the problem? `vendor/bin/phing tests-fast` (which is mentioned in the README) runs all essential tests in around 14 seconds on my computer.

Reply:
Ok, that was probably a problem on my end. `tests-fast` took about 20 minutes in my PHP 7.4-cli docker container. Probably something with xdebug...

Thanks for your help and this nice peace of software! :octocat:

Reply:
Yeah, Xdebug will slow it down, I enable it only when I need to debug something. Thank you!